docs/guides/implement_custom_components.md

```diff
@@ -0,0 +1,171 @@
+
+Lastly, we define custom arguments that the component will support. In our case, we include the `mime_type argument`, which allows us to filter images by different file types in the future.
+
+### Creating the Component
+
+Now, it's time to outline the component logic. To do this, we'll create a `main.py` file within the `components/filter_images/src directory`:
+
+
+```python
+"""A component that downloads common crawl files."""
+import logging
+import mimetypes
+import pandas as pd
+from fondant.component import PandasTransformComponent
+
+logger = logging.getLogger(__name__)
+
+
+class FileTypeFilter(PandasTransformComponent):
+    """Custom component to filter on specific file type based on url"""
+
+    def __init__(self, *args, mime_type: str):
+        self.mime_type = mime_type
+
+    @staticmethod
+    def get_mime_type(data):
+        """Guess mime type based on the file name"""
+        mime_type, _ = mimetypes.guess_type(data)
+        return mime_type
+
+    def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
+        """Reduce dataframe to specific mime type"""
+        dataframe[("images", "mime_type")] = dataframe[("images", "url")].apply(
+            self.get_mime_type
+        )
+
+        return dataframe[dataframe[("images", "mime_type")] == self.mime_type]
+```
+
+
+
+By doing this, we create a custom component that inherits from a `PandasTransformComponent`. This specialised component works with pandas dataframes, allowing us the freedom to modify them as needed before returning the resulting dataframe.
+In this particular example, our component guesses the MIME type of each image based on its URL. Subsequently, it adds this information to the `images` subset of the dataframe and returns the filtered dataset based on the desired MIME type.
+
+
+## Build the component
+
+
+To use the component, Fondant must package it into an executable Docker image. To achieve this, we need to define a Dockerfile. You can create this file within the `components/filter_images` folder using the following content:
+
+```
+FROM --platform=linux/amd64 python:3.8-slim as base
+
+# System dependencies
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install git -y
+
+# Install requirements
+COPY requirements.txt /
+RUN pip3 install --no-cache-dir -r requirements.txt
+
+# Install Fondant
+# This is split from other requirements to leverage caching
+ARG FONDANT_VERSION=main
+RUN pip3 install fondant[aws,azure,gcp]@git+https://github.com/ml6team/fondant@main
+
+# Set the working directory to the component folder
+WORKDIR /component/src
+
+# Copy over src-files
+COPY src/ .
+
+ENTRYPOINT ["fondant", "execute", "main"]
+```
+
+As part of the Dockerfile build process, we install necessary dependencies. Consequently, we must create a `requirements.txt` file in the `components/filter_images` folder. If your components logic demands custom libraries, you can include them in the requirements.txt file.
+
```

Comment:
just saw it but we actually forgot to show the requirements.txt file. + shouldn't this section must be a subsection as well ? --> ### Build the component